internal/common/simulation/ode.go

```diff
@@ -0,0 +1,76 @@
+package simulation
+
+import (
+	"fmt"
+	"math"
+)
+
+type ypFunc func(t, y float64) float64         // dy/dt definition
+type ypStepFunc func(t, y, dt float64) float64 // step function for computing dy/dt
+
+// newRKStep takes a function representing an ODE
+// and returns a function that performs a single step of the 4th order
+// Runge-Kutta method. See https://en.wikipedia.org/wiki/Runge–Kutta_methods
+func newRK4Step(yp ypFunc) ypStepFunc {
```

Comment:
https://pkg.go.dev/github.com/ready-steady/ode/rk4

i reckon there are good packages out there for rk?

Reply:
I did see one or two but they didn't seem fantastic - prob could've persevered a little more. Was kind of just interested to try myself haha. Will think about this one though. Could just define dy/dt = 2t and then check that we get y = t^2. Obviously not exhaustive but yeah

Reply:
Mad props for doing ODE's by hand. I swear that was like an assignment by itself a couple years back hahaha